### PR TITLE
[US3437] feat(mayactl): add pool list and describe command to mayactl.

### PR DESCRIPTION
**Pull request by ashishranjan738**
**What this PR does / why we need it**:
-> This will add pool list and describe command to mayactl.

**Special notes for your reviewer**:
Output of pool list:
```
POOL NAME                  NODE NAME     POOL TYPE
---------                  ---------     ---------
cstor-sparse-pool-qte1     127.0.0.1     striped
```
Output of pool describe:
```
Pool Details :
--------------
Storage Pool Name  : cstor-sparse-pool-qte1
Node Name          : 127.0.0.1
CAS Template Used  : cstor-pool-create-default-0.8.0
CAS Type           : cstor
StoragePoolClaim   : cstor-sparse-pool
UID                : b5a62c11-e8eb-11e8-9ec2-b4b686bd0cff
Pool Type          : striped
Over Provisioning  : false

Disk List :
----------
sparse-5a92ced3e2ee21eac7b930f670b5eab5
```

### CODE REVIEW

File: pkg/install/v1alpha1/storage_pool.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+apiVersion: openebs.io/v1alpha1
+kind: RunTask
+metadata:
+  name: storage-pool-read-default
+spec: 
+  meta: |
+    id: readstoragepool
+    apiVersion: openebs.io/v1alpha1
+    kind: StoragePool
+    action: list
```
Comment (AmitKumarDas) [inline pkg/install/v1alpha1/storage_pool.go]: why are we using list? Why not read?

Comment (ashishranjan738) [reply pkg/install/v1alpha1/storage_pool.go]: changed to get action @AmitKumarDas .

File: cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+	glog.Infof("Cas template based pools were listed successfully")
+	return pools, nil
+}
+
+func (p *poolAPIOpsV1alpha1) read(poolName string) (*v1alpha1.StoragePool, error) {
+	glog.Infof("CAS template based storage pool read request was received")
+
+	sOps, err := pool.NewStoragePoolOperation(poolName)
+	if err != nil {
```
Comment (AmitKumarDas) [inline cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go]: Please check for `IsNotFound` error & return that as 404.
Please refer to similar handling in other endpoint files.

File: cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		return nil, CodedError(400, err.Error())
+	}
+
+	pools, err := sOps.Read()
+	if err != nil {
```
Comment (AmitKumarDas) [inline cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go]: Please check for `IsNotFound` error & return that as 404.
Please refer to similar handling in other endpoint files.

File: cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	glog.Infof("Cas template based pools were listed successfully")
+	return pools, nil
+}
+
+func (p *poolAPIOpsV1alpha1) read(poolName string) (*v1alpha1.StoragePool, error) {
+	glog.Infof("CAS template based storage pool read request was received")
+
+	sOps, err := pool.NewStoragePoolOperation(poolName)
+	if err != nil {
+		return nil, CodedError(400, err.Error())
```
Comment (AmitKumarDas) [inline cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go]: Why should this be a 400 error?

Comment (ashishranjan738) [reply cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go]: sure will change it.

File: cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go (hunk 3 above)
Comment (ashishranjan738) [reply cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go]: sure @AmitKumarDas

Comment (ashishranjan738) [reply cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go]: I think it should be Internal server error 500 ?
Any suggestions @AmitKumarDas  ?

File: pkg/install/v1alpha1/storage_pool.go (hunk 1 above)
Comment (ashishranjan738) [reply pkg/install/v1alpha1/storage_pool.go]: ok going with 404. Reference : https://stackoverflow.com/questions/5604816/whats-the-most-appropriate-http-status-code-for-an-item-not-found-error-page

File: cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go (hunk 3 above)
Comment (ashishranjan738) [reply cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go]: ok going with 404. Reference : https://stackoverflow.com/questions/5604816/whats-the-most-appropriate-http-status-code-for-an-item-not-found-error-page

File: cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go (hunk 4 above)
Comment (ashishranjan738) [reply cmd/maya-apiserver/app/server/pool_endpoint_v1alpha1.go]: ok going with 404. Reference : https://stackoverflow.com/questions/5604816/whats-the-most-appropriate-http-status-code-for-an-item-not-found-error-page

**Pull request merged**